### PR TITLE
H-291: Add Python package for `hash-graph-client`

### PR DESCRIPTION
**Pull request by indietyp**
## 🌟 What is the purpose of this PR?

Create the package skeleton for a Python equivalent of `hash-graph-client` for a semi-automatically generated Python client from the OpenAPI specification.

The first attempt (not in history) was to merge both the typescript and python language in the same directory, while possible it was **very** confusing and would be hard to maintain. To combat this, the typescript client has been moved into `typescript/`, while the `python` client lives in the `python/` subdirectory.

The `typescript` package remains with the `@local/hash-graph-client` package name, while the python package uses `@local/hash-graph-client-python`, but(!) the official name (in `pyproject.toml`) is `hash-graph-client`.

As discussed with @nonparibus offline, the package is to be licensed w/ AGPL.

> The move and other possibilities of package arrangements have been discussed offline with @TimDiekmann.

## 🔍 What does this change?

* add `hash-graph-client` python package.

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] affected the execution graph, and the `turbo.json`'s have been updated to reflect this

## 🐾 Next steps

* generate client, and move client over from internal repo

### CODE REVIEW

File: libs/@local/hash-graph-client/python/turbo.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "extends": ["//"],
+  "pipeline": {
+    "build": {
+      "dependsOn": ["^build", "codegen"],
+      "inputs": ["./**/*.py", "pyproject.toml", "poetry.lock", "LICENSE*"],
+      "outputs": ["dist/**"]
+    },
```
Comment (TimDiekmann) [inline libs/@local/hash-graph-client/python/turbo.json:8]: Does this not depend on the OpenAPI?

File: libs/@local/hash-graph-client/typescript/openapitools.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
   "spaces": 2,
   "generator-cli": {
     "version": "6.6.0",
     "generators": {
       "ts": {
         "generateAliasAsModel": true,
         "generatorName": "typescript-axios",
-        "output": "#{cwd}/../hash-graph-client",
-        "inputSpec": "../../../apps/hash-graph/openapi/openapi.json",
+        "output": "#{cwd}/../typescript",
```
Comment (TimDiekmann) [inline libs/@local/hash-graph-client/typescript/openapitools.json:10]: Isn't this the same as `"#{cwd}"`?

File: package.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
       "libs/*",
       "libs/@hashintel/*",
       "libs/@local/*",
+      "libs/@local/*/typescript",
```
Comment (TimDiekmann) [inline package.json:11]: We want the python package to be part of the workspace as well.
```suggestion
      "libs/@local/*/typescript",
      "libs/@local/*/python",
```

File: libs/@local/hash-graph-client/typescript/openapitools.json (hunk 2 above)
Comment (indietyp) [reply libs/@local/hash-graph-client/typescript/openapitools.json:10]: I thought the same but was anxious to change the previously configured value (which also points to `#{cwd}`). I'll change it an see what happens.

File: libs/@local/hash-graph-client/python/turbo.json (hunk 1 above)
Comment (indietyp) [reply libs/@local/hash-graph-client/python/turbo.json:8]: Not yet, #2842 adds that dependency

File: libs/@local/hash-graph-client/typescript/openapitools.json (hunk 2 above)
Comment (indietyp) [reply libs/@local/hash-graph-client/typescript/openapitools.json]: fixed in [4367f05](https://github.com/hashintel/hash/pull/2841/commits/4367f052c76d88b83b2be7386cc6db1d5abe57e2)

**Review by TimDiekmann — DISMISSED** [review]

**Review by nonparibus — APPROVED** [review]

File: libs/@local/hash-graph-client/typescript/turbo.json
Comment (TimDiekmann) [inline libs/@local/hash-graph-client/typescript/turbo.json:1]: The input was supposed to be changed here 😢

**Pull request merged**